src/Microsoft.Diagnostics.Monitoring/DiagnosticServices.cs

```diff
@@ -0,0 +1,118 @@
+﻿using Microsoft.Diagnostics.Monitoring.RestServer;
+using Microsoft.Diagnostics.NETCore.Client;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring
+{
+    public sealed class DiagnosticServices : IDiagnosticServices
+    {
+        private const int MaxTraceSeconds = 60 * 5;
+
+        private readonly ILogger<DiagnosticsMonitor> _logger;
+        private readonly CancellationTokenSource _tokenSource = new CancellationTokenSource();
+
+        public DiagnosticServices(ILogger<DiagnosticsMonitor> logger)
+        {
+            _logger = logger;
+        }
+
+        public IEnumerable<int> GetProcesses()
+        {
+            //TODO This won't work properly with multi-container scenarios that don't share the process space.
+            //TODO We will need to use DiagnosticsAgent if we are the server.
+            return DiagnosticsClient.GetPublishedProcesses();
+        }
+
+        public async Task<Stream> GetDump(int pid, DumpType mode)
+        {
+            string dumpFilePath = FormattableString.Invariant($@"{Path.GetTempPath()}{Path.DirectorySeparatorChar}{Guid.NewGuid()}_{pid}");
+            NETCore.Client.DumpType dumpType = MapDumpType(mode);
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                // Get the process
+                Process process = Process.GetProcessById(pid);
+                await Dumper.CollectDumpAsync(process, dumpFilePath, dumpType);
+            }
+            else
+            {
+                await Task.Run(() =>
```

Comment:
Do we need to Task.Run this as well? @noahfalk is there no async version of this API?

Reply:
None of the diagnostics client APIs are async yet.